packages/@yoda/bluetooth/hfp.js

```diff
@@ -0,0 +1,275 @@
+'use strict'
+
+var logger = require('logger')('bluetooth-hfp')
+var inherits = require('util').inherits
+var EventEmitter = require('events').EventEmitter
+var floraFactory = require('@yoda/flora')
+var FloraComp = require('@yoda/flora/comp')
+var stateFilters = require('./hfp-statemap').stateFilters
+
+/**
+ * @typedef {object} PROFILE
+ * @property {string} BLE - Bluetooth low energy profile.
+ * @property {string} A2DP - Bluetooth advanced audio distribution profile.
+ * @property {string} HFP - Bluetooth hands-free profile.
+ */
+
+/**
+ * @typedef {object} RADIO_STATE
+ * @property {string} ON - Bluetooth radio state is ON.
+ * @property {string} OFF - Bluetooth radio state is OFF.
+ * @property {string} ON_FAILED - Turn on bluetooth radio failed.
+ */
+
+/**
+ * @typedef {object} CONNECTION_STATE
+ * @property {string} CONNECTED - Successfully connected to remote device.
+ * @property {string} DISCONNECTED - Disconnected from remote deivce.
+ * @property {string} CONNECT_FAILED - Failed to connect to remote device.
+ * @property {string} AUTOCONNECT_FAILED - Auto connection to history paired device failed after turn on bluetooth.
+ */
+
+/**
+ * When bluetooth's radio state is changed, such as on/off.
+ * @event module:@yoda/bluetooth.BluetoothHfp#radio_state_changed
+ */
+
+/**
+ * When bluetooth's connection state is changed, such as connected/disconnected.
+ * @event module:@yoda/bluetooth.BluetoothHfp#connection_state_changed
+ */
+
+/**
+ * When bluetooth's call state is changed, such as idle/incoming.
+ * @event module:@yoda/bluetooth.BluetoothHfp#call_state_changed
+ */
+
+/**
+ * Use `bluetooth.getAdapter(PROFILE.HFP)` instead of this constructor.
+ * @memberof module:@yoda/bluetooth
+ * @constructor
+ * @extends EventEmitter
+ * @param {string} deviceName - The device name.
+ */
+function BluetoothHfp (deviceName) {
```

Comment:
Now we could use the keywords `class` and `extends`.

Reply:
Changed to class extends EventEmitter.